sqlstats/sqlstats_test.go

```diff
@@ -119,3 +121,62 @@ func TestCollect(t *testing.T) {
 		}
 	}
 }
+
+func TestTracerResetRace(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	tracer := &Tracer{}
+	db := sql.OpenDB(sqlite.Connector("file:"+t.TempDir()+"/test.db", nil, tracer))
+	defer db.Close()
+
+	tx, err := db.BeginTx(ctx, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tx.Rollback()
+	if _, err := tx.ExecContext(ctx, "CREATE TABLE t (c);"); err != nil {
+		t.Fatal(err)
+	}
+	if err := tx.Commit(); err != nil {
+		t.Fatal(err)
+	}
+
+	// Continually reset the tracer.
+	go func() {
+		for ctx.Err() == nil {
+			tracer.Reset()
+		}
+	}()
+
+	// Continually grab transactions and insert into the database.
+	errc := make(chan error, 1)
+	go func() {
+		defer close(errc)
```

Comment:
Was the panic check not desired?

Reply:
Not undesirable - just unnecessary, as the test will fail there's a panic in any of these goroutines. I added it originally because for some reason I thought the default behaviour was to let the goroutine die quietly, but it crashes the whole program.